pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -36,13 +38,43 @@ const (
 )
 
 type StorageInitializerConfig struct {
-	Image string `json:"image"`
+	Image         string `json:"image"`
+	CpuRequest    string `json:"cpuRequest"`
+	CpuLimit      string `json:"cpuLimit"`
+	MemoryRequest string `json:"memoryRequest"`
+	MemoryLimit   string `json:"memoryLimit"`
 }
+
 type StorageInitializerInjector struct {
 	credentialBuilder *credentials.CredentialBuilder
 	config            *StorageInitializerConfig
 }
 
+func getStorageInitializerConfigs(configMap *v1.ConfigMap) (*StorageInitializerConfig, error) {
+	storageInitializerConfig := &StorageInitializerConfig{}
+	if initializerConfig, ok := configMap.Data[StorageInitializerConfigMapKeyName]; ok {
+		err := json.Unmarshal([]byte(initializerConfig), &storageInitializerConfig)
+		if err != nil {
+			panic(fmt.Errorf("Unable to unmarshall %v json string due to %v ", StorageInitializerConfigMapKeyName, err))
+		}
+	}
+
+	if storageInitializerConfig.MemoryRequest == "" {
+		storageInitializerConfig.MemoryRequest = StorageInitializerDefaultMemoryRequest
```

Comment:
call `MustParse` here to make sure all the limit values are sane?

Reply:
Good point! I think we can use ParseQuantity and check the error. MustParse will panic lol